receiver/statsdreceiver/protocol/statsd_parser.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protocol
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+// StatsDParser supports the Parse method for parsing StatsD messages with Tags.
+type StatsDParser struct{}
+
+// Parse returns an OTLP metric representation of the input StatsD string.
+func (p *StatsDParser) Parse(line string) (*metricspb.Metric, error) {
+	parts := strings.Split(line, "|")
```

Comment:
Might consider using `IndexByte` and slices instead of split since number of parts seems to be fixed for both splits

Reply:
The number of split parts for the split by `|` will be variable as more types are added due to some types supporting `@<sample-rate>` and `#<label-key/values>` being optional. The second split by `:` **should** be fixed for properly formed messages, so I can look to use it there. Is the idea to get the index and use that to slice out the values?

Reply:
Yeah that's what I was thinking